### PR TITLE
Add a filter('editor.FeaturedImage') for the FeaturedImage component

### PR DESCRIPTION
**Pull request by adamsilverstein**
## Description

Enables developers to change what is displayed for the post's featured image panel, or append to it.

Fixes https://github.com/WordPress/gutenberg/issues/6533.

Similar to the current core `admin_post_thumbnail_html` filter, this enables filtering of the featured image component.

## How has this been tested?
Using the console, I tested replacing and appending to the panel.

#### Replacing the panel
```js
// Replace the panel.
wp.hooks.addFilter( 
	'editor.PostFeaturedImage', 
	'myhook', 
	function() { 
		return function() { 
			return wp.element.createElement( 
				'div', 
				{}, 
				'This is a test' 
			); 
		} 
	} 
);
```

#### Appending to the panel
```js
wp.hooks.addFilter( 
	'editor.PostFeaturedImage', 
	'myhook', 
	function( original ) { 
		return function() { 
			return (
				wp.element.createElement( 
					'div', 
					{ key: 'outer' + Math.random() }, 
					[
						_.extend( original( {} ), { key: 'below' + Math.random() } ),
						'Add something below...' 
					]
				)
			);
		} 
	} 
);
```

## Screenshots 
replace

append

## Types of changes
Filters the FeaturedImage component.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation.

### CODE REVIEW

**Comment by danielbachhuber** [comment]
@adamsilverstein @gziolo Prior to merging, can one of you open a PR against https://github.com/danielbachhuber/gutenberg-migration-guide/blob/master/filter-admin-post-thumbnail-html.md with documentation on how this works?

**Review by iwtw-richb — APPROVED** [review]

**Review by gziolo — COMMENTED** [review]
Did you consider adding this filter directly on the editor component instead of on the panel wrapper?

My concern is that you rather don’t want to allow plugin developers to replace this panel with anything else. You rather would expect to replace panel’s content with something else, prepend or append HTML code or disable the panel completely.

I think that we should rather tackle removing individual panels separately and focus here on the Featured Image itself

**Comment by adamsilverstein** [comment]
> Did you consider adding this filter directly on the editor component instead of on the panel wrapper?

Good suggestion, I’ll give that a try.

**Comment by adamsilverstein** [comment]
> Did you consider adding this filter directly on the editor component instead of on the panel wrapper?

@gziolo I did this in https://github.com/WordPress/gutenberg/pull/6940/commits/77a4366a2d1dee90df2dab4598bb888e276231a6 and this fixes the issues I noted originally. Let me know what you think and if it looks good I can try to update docs as well.

**Review by gziolo — COMMENTED** [review]
All good, it’s exactly what I thought about. Let’s add docs and get it in!

**Comment by adamsilverstein** [comment]
@danielbachhuber opened a PR - https://github.com/danielbachhuber/gutenberg-migration-guide/pull/8

@gziolo took a pass at adding a readme for the component, is this what you had in mind?

**Review by gziolo — APPROVED** [review]
@adamsilverstein thanks for working on this one, I updated docs to follow other examples. In other files we use ES.Next but I personally prefer to have all the examples using filters to use ES5 so you could paste them in the JS console.

File: editor/components/post-featured-image/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+```js
+wp.hooks.addFilter( 
+	'editor.PostFeaturedImage', 
+	'myplugin/myhook', 
+	function( original ) { 
+		return function() { 
+			return (
+				wp.element.createElement( 
+					'div', 
+					{ key: 'outer' + Math.random() },
```
[the remark this answers is not in the document]
Comment (adamsilverstein) [reply editor/components/post-featured-image/README.md]: right, that was bad! thanks for fixing that and cleanup generally.

**Comment by adamsilverstein** [comment]
> I personally prefer to have all the examples using filters to use ES5 so you could paste them in the JS console.

For sure that is great, thanks for updating those examples!

**Pull request merged**